doc/security_need_to_know.md

```diff
@@ -15,19 +15,45 @@ It is commonly stated that a security issue is an issue that:
 
 Given that WASI is a set of Capability-based APIs, all unauthorized actions are not supposed to happen. Most of the above security concerns can be alleviated. What remains for us is to ensure that the execution of Wasm modules is secure. In other words, do not compromise the sandbox. Unless it is explicitly disabled beforehand.
 
-Thus, we share most of the criteria for judging security issues with [the Bytecode Alliance](https://github.com/bytecodealliance/rfcs/blob/main/accepted/what-is-considered-a-security-bug.md#definition).
+### Is this bug considered a security vulnerability?
+
+For someone who finds a problem, if a bug **results in crash or hang**, please treat it as a security problem and report it to a security advisor. The maintainer will look into it and change its category if needed. It is better safe than sorry.
+
+If the author of an issue(results in crash or hang) can go through the following checklist and answer all questions with "No", it is fine to mark it as a regular bug. If not, please report it as a security issue.
+
+---
+
+For those maintainers, please use the following guidelines to determine if a bug or advisory is a security issue:
+
+Only bugs that affect [tier A platforms or features](./tired_support.md) should be considered.
+
+Actions that differ from Wasm rules (like calculating wrong values) are not seen as security issues as long as they stay within the sandbox.
+
+As embedded code, trusts inputs from the guest. By default, APIs and CLIs are following the principle of **caller guarantee**. If the caller provides incorrect parameters, it is not a security issue. For example, if a user passes an invalid file descriptor to `fd_read`, it is not a security issue.
```

Comment:
If following the caller guarantee rule, several corner cases should be explicitly identified. 
- How should we handle the following cases (and possibly more) in function parameters?
  - Null pointer
  - Empty string
  - Empty list/set/array
  - Meaningless zero

- Should we expand the "caller guarantee" rule to iwasm and wamrc command line options?